sql/2021/pwa/manifests_and_service_workers.sql

```diff
@@ -0,0 +1,22 @@
+#standardSQL
+# Counting Manifests and Service Workers
+SELECT
+  client,
+  SAFE_DIVIDE(SUM(ServiceWorker), SUM(COUNT(0)) OVER (PARTITION BY client)) AS serviceworkers_pct,
+  SAFE_DIVIDE(SUM(manifests), SUM(COUNT(0)) OVER (PARTITION BY client)) AS manifests_pct,
+  SAFE_DIVIDE(COUNTIF(ServiceWorker > 0 OR manifests > 0), SUM(COUNT(0)) OVER (PARTITION BY client)) AS either_pct,
+  SAFE_DIVIDE(COUNTIF(ServiceWorker > 0 AND manifests > 0), SUM(COUNT(0)) OVER (PARTITION BY client)) AS both_pct,
+  SUM(COUNT(0)) OVER (PARTITION BY client) AS total
+FROM (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    IF(JSON_EXTRACT(payload, '$._pwa') != "[]" AND JSON_EXTRACT(payload, '$._pwa.serviceWorkers') != "[]",1,0) AS ServiceWorker,
```

Comment:
The test run shows a very low number for the present of each and also of both. This seems to be consistent with [2020](https://almanac.httparchive.org/en/2020/pwa#web-app-manifests). Once again it would be interesting to check YoY on growth.
Here we might also want to consider adding the `$._pwa.serviceWorkerHeuristic` in the final query.